### ISSUE TITLE
Provide capability to deserialize different types of metrics declarations

### ISSUE THREAD

**Issue by tomkerkhove** [tomkerkhove/promitor#129]
Provide capability to deserialize different types of metrics declarations.

Currently blocks from adding more scraper sources.

Related https://github.com/aaubry/YamlDotNet/issues/341, alternative will be to use `YamlStream`.

### Example

```yaml
azureMetadata:
  tenantId: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
  subscriptionId: yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy
  resourceGroupName: promitor
metrics: 
  - name: demo_queue_size
    description: "Amount of active messages of the 'myqueue' queue"
    resourceType: ServiceBusQueue
    namespace: promitor-messaging
    queueName: orders
    azureMetricConfiguration:
      metricName: ActiveMessages
      aggregation: Total 
  - name: demo_generic_metric
    description: "Generic metric information"
    resourceType: Generic
    resourceId: "providers/Microsoft.ServiceBus/namespaces/promitor-messaging"
    oDataFilter: EntityName eq 'orders'
    azureMetricConfiguration:
      metricName: ActiveMessages
      aggregation: Total
```

**Comment by tomkerkhove** [comment]
Currently POCed here - https://github.com/tomkerkhove/yaml-generics and waiting on https://github.com/aaubry/YamlDotNet/issues/341 for clarification

**Comment by tomkerkhove** [comment]
Closed in #149
